File: samples/Sample_S100/VB.NET_SBXBPCSample/frmMain.vb
Option Strict Off
Option Explicit On
Friend Class frmMain
	Inherits System.Windows.Forms.Form
	Public gMachineNumber As Integer
	Dim mOpenFlag As Boolean
	
	'UPGRADE_WARNING: Event cmbMachineNumber.SelectedIndexChanged may fire when form is initialized. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="88B12AE1-6DE0-48A0-86F1-60C0686C026A"'
	Private Sub cmbMachineNumber_SelectedIndexChanged(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmbMachineNumber.SelectedIndexChanged
		gMachineNumber = cmbMachineNumber.SelectedIndex + 1
	End Sub
	
	Private Sub cmdBellInfo_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdBellInfo.Click
		Me.Visible = False
		frmBellInfo.Visible = True
	End Sub
	
	Private Sub cmdClose_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdClose.Click
		If mOpenFlag = True Then
            SBXPC1.Disconnect()
            mOpenFlag = False
            cmdOpen.Enabled = True
            cmdClose.Enabled = False
            EnableManagementGroup(False)
        End If
    End Sub

    Private Sub cmdEnrollData_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdEnrollData.Click
        Me.Visible = False
        frmEnroll.Visible = True
    End Sub

    Private Sub cmdExit_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdExit.Click
        Me.Close()
    End Sub

    Private Sub cmdLockCtl_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdLockCtl.Click
        Me.Visible = False
        frmLockCtl.Visible = True
    End Sub

    Private Sub cmdLogData_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdLogData.Click
        Me.Visible = False
        frmLog.Visible = True
    End Sub

    Private Sub cmdOpen_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdOpen.Click
        Dim lpszIPAddress As String

        If optNetworkDevice.Checked = True Then
            lpszIPAddress = txtIPAddress.Text
            If SBXPC1.ConnectTcpip(gMachineNumber, lpszIPAddress, CInt(txtPortNo.Text), CInt(txtPassword.Text)) = True Then
                mOpenFlag = True
                cmdOpen.Enabled = False
                cmdClose.Enabled = True
                EnableManagementGroup(True)
            End If
        End If
        If optSerialDevice.Checked = True Then
            If SBXPC1.ConnectSerial(gMachineNumber, cmbComPort.SelectedIndex + 1, CInt(cmbBaudrate.Text)) = True Then
                mOpenFlag = True
                cmdOpen.Enabled = False
                cmdClose.Enabled = True
                EnableManagementGroup(True)
            End If
        End If
        If optUSBDevice.Checked = True Then
            If SBXPC1.ConnectSerial(gMachineNumber, 0, 0) = True Then
                mOpenFlag = True
                cmdOpen.Enabled = False
                cmdClose.Enabled = True
                EnableManagementGroup(True)
            End If
        End If
    End Sub

    Private Sub cmdProductCode_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdProductCode.Click
        Me.Visible = False
        frmSerialNo.Visible = True
    End Sub

    Private Sub cmdSystemInfo_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdSystemInfo.Click
        Me.Visible = False
        frmSystemInfo.Visible = True
    End Sub

    Private Sub frmMain_Load(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles MyBase.Load
        lblComPort.Enabled = False
        cmbComPort.Enabled = False
        lblBaudrate.Enabled = False
        cmbBaudrate.Enabled = False

        optNetworkDevice.Checked = True
        lblIPAddress.Enabled = True
        txtIPAddress.Enabled = True
        lblPortNo.Enabled = True
        txtPortNo.Enabled = True
        lblPassword.Enabled = True
        txtPassword.Enabled = True

        cmdOpen.Enabled = True
        cmdClose.Enabled = False
        EnableManagementGroup(False)
        mOpenFlag = False
        cmbMachineNumber.Text = CStr(1)
        cmbComPort.Text = CStr(1)
        cmbBaudrate.Text = "115200"

        SBXPC1.DotNET()
    End Sub

    Private Sub EnableManagementGroup(ByVal bEnable As Boolean)
        cmdEnrollData.Enabled = bEnable
        cmdLogData.Enabled = bEnable
        cmdSystemInfo.Enabled = bEnable
        cmdProductCode.Enabled = bEnable
        cmdBellInfo.Enabled = bEnable
        cmdLockCtl.Enabled = bEnable
        cmdTrMode.Enabled = bEnable
        cmdAccessTz.Enabled = bEnable
        cmdModeTZone.Enabled = bEnable
        cmdHoliday.Enabled = bEnable
    End Sub

    Private Sub frmMain_FormClosed(ByVal eventSender As System.Object, ByVal eventArgs As System.Windows.Forms.FormClosedEventArgs) Handles Me.FormClosed
        If mOpenFlag = True Then
            SBXPC1.Disconnect()
            mOpenFlag = False
        End If
    End Sub

    'UPGRADE_WARNING: Event optNetworkDevice.CheckedChanged may fire when form is initialized. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="88B12AE1-6DE0-48A0-86F1-60C0686C026A"'
    Private Sub optNetworkDevice_CheckedChanged(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles optNetworkDevice.CheckedChanged
        If eventSender.Checked Then
            Dim lpszIPAddress As String
            optSerialDevice.Checked = False
            optUSBDevice.Checked = False

            If optNetworkDevice.Checked = True Then
                lblComPort.Enabled = False
                cmbComPort.Enabled = False
                lblBaudrate.Enabled = False
                cmbBaudrate.Enabled = False
                lblIPAddress.Enabled = True
                txtIPAddress.Enabled = True
                lblPortNo.Enabled = True
                txtPortNo.Enabled = True
                lblPassword.Enabled = True
                txtPassword.Enabled = True
                lpszIPAddress = txtIPAddress.Text
            ElseIf optSerialDevice.Checked = True Then
                lblComPort.Enabled = True
                cmbComPort.Enabled = True
                lblBaudrate.Enabled = True
                cmbBaudrate.Enabled = True
                lblIPAddress.Enabled = False
                txtIPAddress.Enabled = False
                lblPortNo.Enabled = False
                txtPortNo.Enabled = False
                lblPassword.Enabled = False
                txtPassword.Enabled = False
            Else
                lblComPort.Enabled = False
                cmbComPort.Enabled = False
                lblBaudrate.Enabled = False
                cmbBaudrate.Enabled = False
                lblIPAddress.Enabled = False
                txtIPAddress.Enabled = False
                lblPortNo.Enabled = False
                txtPortNo.Enabled = False
                lblPassword.Enabled = False
                txtPassword.Enabled = False
            End If
        End If
    End Sub

    'UPGRADE_WARNING: Event optSerialDevice.CheckedChanged may fire when form is initialized. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="88B12AE1-6DE0-48A0-86F1-60C0686C026A"'
    Private Sub optSerialDevice_CheckedChanged(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles optSerialDevice.CheckedChanged
        If eventSender.Checked Then
            Dim lpszIPAddress As String
            optNetworkDevice.Checked = False
            optUSBDevice.Checked = False

            If optSerialDevice.Checked = True Then
                lblComPort.Enabled = True
                cmbComPort.Enabled = True
                lblBaudrate.Enabled = True
                cmbBaudrate.Enabled = True
                lblIPAddress.Enabled = False
                txtIPAddress.Enabled = False
                lblPortNo.Enabled = False
                txtPortNo.Enabled = False
                lblPassword.Enabled = False
                txtPassword.Enabled = False
            ElseIf optNetworkDevice.Checked = True Then
                lblComPort.Enabled = False
                cmbComPort.Enabled = False
                lblBaudrate.Enabled = False
                cmbBaudrate.Enabled = False
                lblIPAddress.Enabled = True
                txtIPAddress.Enabled = True
                lblPortNo.Enabled = True
                txtPortNo.Enabled = True
                lblPassword.Enabled = True
                txtPassword.Enabled = True
                lpszIPAddress = txtIPAddress.Text
            Else
                lblComPort.Enabled = False
                cmbComPort.Enabled = False
                lblBaudrate.Enabled = False
                cmbBaudrate.Enabled = False
                lblIPAddress.Enabled = False
                txtIPAddress.Enabled = False
                lblPortNo.Enabled = False
                txtPortNo.Enabled = False
                lblPassword.Enabled = False
                txtPassword.Enabled = False
            End If
        End If
    End Sub

    Private Sub optUSBDevice_CheckedChanged(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles optUSBDevice.CheckedChanged
        If eventSender.Checked Then
            Dim lpszIPAddress As String
            optSerialDevice.Checked = False
            optNetworkDevice.Checked = False

            If optSerialDevice.Checked = True Then
                lblComPort.Enabled = True
                cmbComPort.Enabled = True
                lblBaudrate.Enabled = True
                cmbBaudrate.Enabled = True
                lblIPAddress.Enabled = False
                txtIPAddress.Enabled = False
                lblPortNo.Enabled = False
                txtPortNo.Enabled = False
                lblPassword.Enabled = False
                txtPassword.Enabled = False
            ElseIf optNetworkDevice.Checked = True Then
                lblComPort.Enabled = False
                cmbComPort.Enabled = False
                lblBaudrate.Enabled = False
                cmbBaudrate.Enabled = False
                lblIPAddress.Enabled = True
                txtIPAddress.Enabled = True
                lblPortNo.Enabled = True
                txtPortNo.Enabled = True
                lblPassword.Enabled = True
                txtPassword.Enabled = True
                lpszIPAddress = txtIPAddress.Text
            Else
                lblComPort.Enabled = False
                cmbComPort.Enabled = False
                lblBaudrate.Enabled = False
                cmbBaudrate.Enabled = False
                lblIPAddress.Enabled = False
                txtIPAddress.Enabled = False
                lblPortNo.Enabled = False
                txtPortNo.Enabled = False
                lblPassword.Enabled = False
                txtPassword.Enabled = False
            End If
        End If
    End Sub

    'UPGRADE_WARNING: Event txtIPAddress.TextChanged may fire when form is initialized. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="88B12AE1-6DE0-48A0-86F1-60C0686C026A"'
    Private Sub txtIPAddress_TextChanged(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtIPAddress.TextChanged
        Dim lpszIPAddress As String

        If txtIPAddress.Text = "" Then Exit Sub
        If txtPortNo.Text = "" Then Exit Sub
        lpszIPAddress = txtIPAddress.Text
    End Sub

    'UPGRADE_WARNING: Event txtPortNo.TextChanged may fire when form is initialized. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="88B12AE1-6DE0-48A0-86F1-60C0686C026A"'
    Private Sub txtPortNo_TextChanged(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtPortNo.TextChanged
        Dim lpszIPAddress As String

        If txtIPAddress.Text = "" Then Exit Sub
        If txtPortNo.Text = "" Then Exit Sub
        lpszIPAddress = txtIPAddress.Text
    End Sub

    Private Sub cmdTrMode_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdTrMode.Click
        frmTrMode.Visible = True
        Me.Visible = False
    End Sub

    Private Sub cmdAccessTz_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdAccessTz.Click
        frmAccessTz.Visible = True
        Me.Visible = False
    End Sub

    Private Sub cmdModeTZone_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdModeTZone.Click
        frmTMode.Visible = True
        Me.Visible = False
    End Sub

    Private Sub cmdHoliday_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdHoliday.Click
        frmHoliday.Visible = True
        Me.Visible = False
    End Sub
End Class

File: sdk_extracted/20211204-SBXPC-1/samples/Sample_A50/C#_SBXPCSample/SBXPCSampleCSharp/frmMain.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace SBXPCSampleCSharp
{
    public partial class frmMain : Form
    {
        public frmMain()
        {
            InitializeComponent();
        }

        frmEvent frm_event = new frmEvent();

        Boolean mOpenFlag;
        private void cmdOpen_Click(object sender, EventArgs e)
        {
            String lpszIPAddress;
            //		Dim vRet As Boolean
            Program.gMachineNumber = Convert.ToInt32(cmbMachineNumber.Text);
		    if (optNetworkDevice.Checked)
            {
			    lpszIPAddress = txtIPAddress.Text;
			    if (SBXPC1.ConnectTcpip(Program.gMachineNumber, ref lpszIPAddress, Convert.ToInt32(txtPortNo.Text), Convert.ToInt32(txtPassword.Text))) 
                {
				    mOpenFlag = true;
				    cmdOpen.Enabled = false;

				    cmdClose.Enabled = true;
				    cmdEnrollData.Enabled = true;
				    cmdLogData.Enabled = true;
				    cmdSystemInfo.Enabled = true;
				    cmdProuctCode.Enabled = true;
				    cmdBellInfo.Enabled = true;
				    cmdMiscSettings.Enabled = true;
                    cmdAccessTz.Enabled = true;
                    cmdGroupName.Enabled = true;
                    cmdHoliday.Enabled = true;
                    cmdDoorKey.Enabled = true;
                }
		    }
		    if (optSerialDevice.Checked)
            {
			    if (SBXPC1.ConnectSerial(Program.gMachineNumber, cmbComPort.SelectedIndex + 1, Convert.ToInt32(cmbBaudrate.Text)))
                {
				    mOpenFlag = true;
				    cmdOpen.Enabled = false;
				    cmdClose.Enabled = true;
				    cmdEnrollData.Enabled = true;
				    cmdLogData.Enabled = true;
				    cmdSystemInfo.Enabled = true;
				    cmdProuctCode.Enabled = true;
				    cmdBellInfo.Enabled = true;
				    cmdMiscSettings.Enabled = true;
                    cmdAccessTz.Enabled = true;
                    cmdGroupName.Enabled = true;
                    cmdHoliday.Enabled = true;
                    cmdDoorKey.Enabled = true;
                }
		    }
            if (optUSBDevice.Checked)
            {
                if (SBXPC1.ConnectSerial(Program.gMachineNumber, 0, 0))
                {
                    mOpenFlag = true;
                    cmdOpen.Enabled = false;
                    cmdClose.Enabled = true;
                    cmdEnrollData.Enabled = true;
                    cmdLogData.Enabled = true;
                    cmdSystemInfo.Enabled = true;
                    cmdProuctCode.Enabled = true;
                    cmdBellInfo.Enabled = true;
                    cmdMiscSettings.Enabled = true;
                    cmdAccessTz.Enabled = true;
                    cmdGroupName.Enabled = true;
                    cmdHoliday.Enabled = true;
                    cmdDoorKey.Enabled = true;
                }
            }
        }

        public void Reopen(int nMachineNum)
        {
            Program.gMachineNumber = nMachineNum;
            cmbMachineNumber.SelectedIndex = nMachineNum - 1;
            cmdOpen_Click(null, null);
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            optSerialDevice.Checked = false;
            lblComPort.Enabled = false;
            cmbComPort.Enabled = false;
            lblBaudrate.Enabled = false;
            cmbBaudrate.Enabled = false;

            optNetworkDevice.Checked = false;
            lblIPAddress.Enabled = false;
            txtIPAddress.Enabled = false;
            lblPortNo.Enabled = false;
            txtPortNo.Enabled = false;
            lblPassword.Enabled = false;
            txtPassword.Enabled = false;

            cmdOpen.Enabled = true;
            cmdClose.Enabled = false;
            cmdEnrollData.Enabled = false;
            cmdLogData.Enabled = false;
            cmdSystemInfo.Enabled = false;
            cmdProuctCode.Enabled = false;
            cmdBellInfo.Enabled = false;
            cmdMiscSettings.Enabled = false;
            cmdAccessTz.Enabled = false;
            cmdGroupName.Enabled = false;
            cmdHoliday.Enabled = false;
            cmdDoorKey.Enabled = false;

            mOpenFlag = false;
            cmbMachineNumber.Text = Convert.ToString(1);
            cmbComPort.Text = Convert.ToString(1);
            cmbBaudrate.Text = "115200";

            SBXPC1.DotNET();
        }

        private void cmdClose_Click(object sender, EventArgs e)
        {
            if (mOpenFlag == true)
            {
                SBXPC1.Disconnect();
			    mOpenFlag = false;
			    cmdOpen.Enabled = true;
			    cmdClose.Enabled = false;
                cmdEnrollData.Enabled = false;
                cmdLogData.Enabled = false;
                cmdSystemInfo.Enabled = false;
                cmdProuctCode.Enabled = false;
                cmdBellInfo.Enabled = false;
                cmdMiscSettings.Enabled = false;
                cmdAccessTz.Enabled = false;
                cmdGroupName.Enabled = false;
                cmdHoliday.Enabled = false;
                cmdDoorKey.Enabled = false;
            }
        }

        private void optSerialDevice_CheckedChanged(object sender, EventArgs e)
        {
            if (optSerialDevice.Checked)
            {
                String lpszIPAddress;
                optSerialDevice.Checked = true;
                optNetworkDevice.Checked = false;
                optUSBDevice.Checked = false;

                if (optSerialDevice.Checked)
                {
                    lblComPort.Enabled = true;
                    cmbComPort.Enabled = true;
                    lblBaudrate.Enabled = true;
                    cmbBaudrate.Enabled = true;
                    lblIPAddress.Enabled = false;
                    txtIPAddress.Enabled = false;
                    lblPortNo.Enabled = false;
                    txtPortNo.Enabled = false;
                    lblPassword.Enabled = false;
                    txtPassword.Enabled = false;
                }
                else if (optNetworkDevice.Checked)
                {
                    lblComPort.Enabled = false;
                    cmbComPort.Enabled = false;
                    lblBaudrate.Enabled = false;
                    cmbBaudrate.Enabled = false;
                    lblIPAddress.Enabled = true;
                    txtIPAddress.Enabled = true;
                    lblPortNo.Enabled = true;
                    txtPortNo.Enabled = true;
                    lblPassword.Enabled = true;
                    txtPassword.Enabled = true;
                    lpszIPAddress = txtIPAddress.Text;
                }
                else
                {
                    lblComPort.Enabled = false;
                    cmbComPort.Enabled = false;
                    lblBaudrate.Enabled = false;
                    cmbBaudrate.Enabled = false;
                    lblIPAddress.Enabled = false;
                    txtIPAddress.Enabled = false;
                    lblPortNo.Enabled = false;
                    txtPortNo.Enabled = false;
                    lblPassword.Enabled = false;
                    txtPassword.Enabled = false;
                }
            }
        }

        private void optNetworkDevice_CheckedChanged(object sender, EventArgs e)
        {
            if (optNetworkDevice.Checked)
            {
                String lpszIPAddress;
                optSerialDevice.Checked = false;
                optNetworkDevice.Checked = true;
                optUSBDevice.Checked = false;

                if (optSerialDevice.Checked)
                {
                    lblComPort.Enabled = true;
                    cmbComPort.Enabled = true;
                    lblBaudrate.Enabled = true;
                    cmbBaudrate.Enabled = true;
                    lblIPAddress.Enabled = false;
                    txtIPAddress.Enabled = false;
                    lblPortNo.Enabled = false;
                    txtPortNo.Enabled = false;
                    lblPassword.Enabled = false;
                    txtPassword.Enabled = false;
                }
                else if (optNetworkDevice.Checked)
                {
                    lblComPort.Enabled = false;
                    cmbComPort.Enabled = false;
                    lblBaudrate.Enabled = false;
                    cmbBaudrate.Enabled = false;
                    lblIPAddress.Enabled = true;
                    txtIPAddress.Enabled = true;
                    lblPortNo.Enabled = true;
                    txtPortNo.Enabled = true;
                    lblPassword.Enabled = true;
                    txtPassword.Enabled = true;
                    lpszIPAddress = txtIPAddress.Text;
                }
                else
                {
                    lblComPort.Enabled = false;
                    cmbComPort.Enabled = false;
                    lblBaudrate.Enabled = false;
                    cmbBaudrate.Enabled = false;
                    lblIPAddress.Enabled = false;
                    txtIPAddress.Enabled = false;
                    lblPortNo.Enabled = false;
                    txtPortNo.Enabled = false;
                    lblPassword.Enabled = false;
                    txtPassword.Enabled = false;
                }
            }
        }

        private void optUSBDevice_CheckedChanged(object sender, EventArgs e)
        {
            if (optUSBDevice.Checked)
            {
                String lpszIPAddress;
                optSerialDevice.Checked = false;
                optNetworkDevice.Checked = false;
                optUSBDevice.Checked = true;

                if (optSerialDevice.Checked)
                {
                    lblComPort.Enabled = true;
                    cmbComPort.Enabled = true;
                    lblBaudrate.Enabled = true;
                    cmbBaudrate.Enabled = true;
                    lblIPAddress.Enabled = false;
                    txtIPAddress.Enabled = false;
                    lblPortNo.Enabled = false;
                    txtPortNo.Enabled = false;
                    lblPassword.Enabled = false;
                    txtPassword.Enabled = false;
                }
                else if (optNetworkDevice.Checked)
                {
                    lblComPort.Enabled = false;
                    cmbComPort.Enabled = false;
                    lblBaudrate.Enabled = false;
                    cmbBaudrate.Enabled = false;
                    lblIPAddress.Enabled = true;
                    txtIPAddress.Enabled = true;
                    lblPortNo.Enabled = true;
                    txtPortNo.Enabled = true;
                    lblPassword.Enabled = true;
                    txtPassword.Enabled = true;
                    lpszIPAddress = txtIPAddress.Text;
                }
                else
                {
                    lblComPort.Enabled = false;
                    cmbComPort.Enabled = false;
                    lblBaudrate.Enabled = false;
                    cmbBaudrate.Enabled = false;
                    lblIPAddress.Enabled = false;
                    txtIPAddress.Enabled = false;
                    lblPortNo.Enabled = false;
                    txtPortNo.Enabled = false;
                    lblPassword.Enabled = false;
                    txtPassword.Enabled = false;
                }
            }
        }

        private void cmdProuctCode_Click(object sender, EventArgs e)
        {
            frmPrtCode frm_prtcode = new frmPrtCode();
            frm_prtcode.Activate();
            frm_prtcode.Visible = true;
            this.Visible = false;
        }

        private void cmdExit_Click(object sender, EventArgs e)
        {
           if (mOpenFlag) 
               SBXPC1.Disconnect();
            Close();
        }

        private void frmMain_FormClosed(object sender, FormClosedEventArgs e)
        {

        }

        private void cmdSystemInfo_Click(object sender, EventArgs e)
        {
            frmSystemInfo frm_SystemInfo = new frmSystemInfo();
            frm_SystemInfo.Activate();
            frm_SystemInfo.Visible = true;
            this.Visible = false;
        }

        private void cmdMiscSettings_Click(object sender, EventArgs e)
        {
            frmMiscSettings frm_misc = new frmMiscSettings();
            frm_misc.Activate();
            frm_misc.Visible = true;
            this.Visible = false;
        }

        private void cmdBellInfo_Click(object sender, EventArgs e)
        {
            frmBellInfo frm_BellInfo = new frmBellInfo();
            frm_BellInfo.Activate();
            frm_BellInfo.Visible = true;
            this.Visible = false;
        }

        private void cmdLogData_Click(object sender, EventArgs e)
        {
            frmLog frm_log = new frmLog();
            frm_log.Activate();
            frm_log.Visible = true;
            this.Visible = false;
        }

        private void cmdEnrollData_Click(object sender, EventArgs e)
        {
            frmEnroll frm_enroll = new frmEnroll();
            frm_enroll.Activate();
            frm_enroll.Visible = true;
            this.Visible = false;
        }

        private void cmdAccessTz_Click(object sender, EventArgs e)
        {
            frmAccessTz frm_access_tz = new frmAccessTz();
            frm_access_tz.Activate();
            frm_access_tz.Visible = true;
            this.Visible = false;
        }

        private void cmdHoliday_Click(object sender, EventArgs e)
        {
            frmHoliday frm_holiday = new frmHoliday();
            frm_holiday.Activate();
            frm_holiday.Visible = true;
            this.Visible = false;
        }

        private void cmdEventMonitor_Click(object sender, EventArgs e)
        {
            frm_event.Activate();
            frm_event.Visible = true;
            this.Visible = false;
        }

        private void SBXPC1_OnReceiveEventXML(object sender, AxSBXPCLib._DSBXPCEvents_OnReceiveEventXMLEvent e)
        {
            frm_event.ReceiveEvent(e.lpszEventXML);
        }

        private void cmdDoorKey_Click(object sender, EventArgs e)
        {
            frmDoorKey frm_doorKey = new frmDoorKey();
            frm_doorKey.Activate();
            frm_doorKey.Visible = true;
            this.Visible = false;
        }

        private void cmdGroupName_Click(object sender, EventArgs e)
        {
            frmGroup frm_group = new frmGroup();
            frm_group.Activate();
            frm_group.Visible = true;
            this.Visible = false;
        }
    }
}
